advanced/sr_gui_fingertip_visualization/src/sr_gui_fingertip_visualization/dot_unit.py

```diff
@@ -0,0 +1,233 @@
+            self.label[data_field].setText(str(data[data_field]))
+        self._dot.update_color_dot(data['pressure'])
+        self.get_dot().update()
+
+
+class DotUnitBiotacSPMinus(DotUnitGeneric):
+    def __init__(self, parent=None):
+        super().__init__(parent=parent)
+        self._initialize_data_structure()
+        self.resize_dot(20)
+        self._init_widget()
+
+    def _initialize_data_structure(self):
+        self._data_fields = ['pac0', 'pac1', 'pdc', 'tac', 'tdc']
+
+    def _value_to_color(self, value):
+        # to change value (1000 and 200) as its taken from the topic
+        r = min(255, max(0, 255 * (value - 1000) / 200))
+        g = 0
+        b = 255 - r
+        self._dot._color = QColor(r, g, b)
+
+    def _init_widget(self):
+        widget_layout = QFormLayout()
+        widget_layout.setFormAlignment(Qt.AlignRight)
+        widget_layout.setLabelAlignment(Qt.AlignRight)
+
+        widget_layout.addRow(self.get_dot())
+
+        self.data_labels = dict()
+        for data_field in self._data_fields:
+            self.data_labels[data_field] = QLabel("-")
+            self.data_labels[data_field].setMinimumSize(40, 10)
+            self.data_labels[data_field].setSizePolicy(2, 2)
+            widget_layout.addRow(QLabel(data_field+":"), self.data_labels[data_field])
+
+        self.setMinimumSize(100, 200)
+        self.setLayout(widget_layout)
+
+    def update_data(self, data):
+        for data_field in self._data_fields:
+            self.data_labels[data_field].setText(str(data[data_field]))
+        self.get_dot().update_color_dot(data['pdc'])  # to be changed from 'pdc' to the correct value
+        self.get_dot().update()
+
+
+class DotUnitBiotacSPPlus(DotUnitGeneric):
+    def __init__(self, electrode_index, parent=None):
+        super().__init__(index=electrode_index, parent=parent)
+        self._initialize_data_structure(electrode_index)
+        self._init_widget()
+
+    def _initialize_data_structure(self, electrode_index):
+        self.electrode_index = electrode_index
+
+    def _value_to_color(self, value):
+        r = 0.0
+        g = 0.0
+        b = 255.0
+        value = float(value)
+        threshold = (0.0, 1000.0, 2000.0, 3000.0, 4095.0)
+        if value <= threshold[0]:
+            pass
+        elif value < threshold[1]:
+            r = 255
+            g = 255 * (value - threshold[0]) / (threshold[1] - threshold[0])
+            b = 0
+        elif value < threshold[2]:
+            r = 255 * ((threshold[2] - value) / (threshold[2] - threshold[1]))
+            g = 255
+            b = 0
+        elif value < threshold[3]:
+            r = 0
+            g = 255
+            b = 255 * ((value - threshold[2]) / (threshold[3] - threshold[2]))
+        elif value < threshold[4]:
+            r = 0
+            g = 255 * ((threshold[4] - value) / (threshold[4] - threshold[3]))
+            b = 255
+        self._dot._color = QColor(r, g, b)
```

Comment:
you have the function update color, but it is only setting everything to 0, you can update that one and use it